### DISCUSSION TITLE
Welcome to TFT-DSS Discussions!

### DISCUSSION THREAD

**Discussion by Pompeiro** [Pompeiro/TFT-DSS#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by NathanOnera** [comment]
Hello Pompeiro! I have been looking at your project with a lot of attention since i first started thinking of coding my own tft bot. At first my idea was only to learn python/git/machine learning. Basically i wanted to code a program that would take screenshots of opponents board during scouting, and detect how many copies of each champion was out of the pool.

For that matter, i am very interested on the screenshot you have showing your board/bench with the name of the champions on it. That is the first step i am trying to breach before focusing on ennemies bench and board. This step is made difficult by champion movement, making basic detection techniques (scalar product of black and white img matrices) not satisfactory. Machine learning would be an interesting approach, but could be too demanding in terms of real-time computation.

Anyway, feel free to share your thoughts on this topic or any other (although i know you have exams coming in May!)

See you,
Nathan

**Reply by Pompeiro** [reply]
Hi @NathanOnera,

Thanks for a kind words and that you write to me on github. Thats a great source of motivation for me, since I had been burnout a little bit.

The screenshots I did paste to my README on github are coming from NN classifier mostly based on bear detector that you can find in "Practical Deep Learning for Coders - Full Course from fast.ai and Jeremy Howard" [https://www.youtube.com/watch?v=0oyCUWLL_fU](url). In that solution you need to divide screen on multiple rectangles which could contain champion and it tells you what the champion is in that rectangle or is it background.

The next solution is to use YOLO model, I assume I was using YOLO darknet. And I was following tutorial from Roboflow. Real Time Object Detection is far way advanced concept and fits better to gathering information from the game.

I have used python script which were playing TFT and was buying certain type of champions for example brawlers and was taking screenshots of my champions on the field.

The problem with real-time doesn't hurt that much in my opinion because TFT is turn-based.

Also I can recommend for you youtube channel Learn Code By Gaming which inspired me to start this project.

If u want to cooperate with me on this project then the best way to meet me on Discord. Feel free to add me Papaj#3719.

**Comment by dongning-ma** [comment]
I just want to say this project is great. As I was quite infuriated by the game by losing straight from P1 98LP to P3 last season, I was planning to use machine learning to help with my decisions and finally make a fully automated tft bot to play this game like the Alpha Go from DeepMind, so I randomly browse related repos and found this.

Had a quick glance through your project and I have several questions here.
1. I assume currently you are reading the champions on the board using screenshots and computer vision? Have you considered integrating tft APIs from third party such as Overwolf etc. to directly fetch such related information?
2. I assume the current champion data is stored in the csv files and read by the pandas, also do you consider using Riot APIs to fetch those data so as to keep them accurate and up-to-date? (e.g. fetch every time this program starts)
3. The data visualization seems very interesting. In the introduction you showed the visualization is based on your experience in game, i.e., the statistical data from many games? 
4. It seems your bot can also perform some user actions such as buy / level-up etc. By reading the code I assume this is accomplished by capturing the game window and move the cursor and click (which is the black-box way), not the white-box solution, e.g by directly injecting into the game (e.g. by APIs) and make actions?

Again this is really an interesting project!

**Reply by Pompeiro** [reply]
hi @AikawaMafuyu 

1. Currently champion detection doesn't work, I need to update model for current patch. But it was working in set 3 I would say with 80-90% success rate. Also I didn't saw Overwolf API and didn't even consider to use any third party soft.
2. Im getting the data from https://tftactics.gg/ and when there is bigger patch I do update this CSV file. Also I'm using their tierlist to give additional points for champions(DataMining.py file). I will say its not the best solution but works for a long time.
3.  The data that I am presenting on radar plots are just some champs stats scaled from 0 to 1. It helps you to define role or utility for champions. So its only champion stats nothing more.
4. You are right I used black-box way there. But I am trying to get as much information as possible, but to get more stable and correct detections I need to train real time object detection using YOLO.

Feel free to use code for your research or  you can cooperate with me.
